### DISCUSSION TITLE
Re-introduce items falling through broken submarine walls (at least the inner ones)

### DISCUSSION THREAD

**Discussion by NotWendy** [FakeFishGames/Barotrauma#15210] [Ideas and suggestions]
Back in the day, items (anything affected by physics, like an ammo box) used to fall through the broken inner walls of the submarine and get stuck on the broken outer walls (connected to open ocean). And by stuck I mean the items would pass through the outer shell halfway, then stop. If you welded the shell, the items would glitch through it over and over again, with a loud banging noise.

This bug got reported and the issue was "fixed" by preventing ALL walls from letting items pass through when they're fully broken.
It's been, like, a year since that band-aid fix was applied. A cool feature was removed because someone couldn't be bothered to fix the actual bug which was reported, and instead went with the nuclear option.

Maybe we can think about adding that feature back in and addressing the problem with the outer shell collision for real this time?
There are multiple ways in which this could be approached design-wise:
- Disallow items from passing through broken walls which are touching the open ocean, but allow them to pass through broken inner walls. This is how it used to work, barring the collision bug. This is the option I'm personally leaning towards.
- Allow items to pass through ANY broken walls, potentially falling out of the submarine. Pretty hardcore, but dead player bodies do that, so maybe there's something to it. This is the option which got suggested in Discord when I brought up this topic.

**Comment by Ivanzyfer** [comment]
I will never forget the battle with charybdis on the humpback where half of the ammunition boxes lying on the deck fell out through the broken walls of the submarine)
It's a pity that it was removed

**Comment by Xeavas** [comment]
I agree with the first point to have it return to how it used to work. (Allow internal pass through but don't allow pass through into ocean)

**Comment by NotWendy** [comment]
After some yapping in baro-unstable, Xeavas and I found out that items also can't pass into the submarine from the outside, unless they're being carried by someone. It would be nice if they were always allowed to pass through doors and hatches, so you could, say, drop something into an open airlock hatch, or toss an item into an open airlock door. Or even scoop items up with your submarine, in some situations.

**Comment by A2ur3** [comment]
#14054
